soc/arm/nordic_nrf/nrf51/Kconfig.series

```diff
@@ -12,7 +12,7 @@ config  SOC_SERIES_NRF51X
 	select SOC_FAMILY_NRF
 	select HAS_SYS_POWER_STATE_DEEP_SLEEP_1
 	select XIP
-	select HAS_CMSIS
+	select HAS_CMSIS_CORE
```

Comment:
Could we have the same thing for Nordic, as for the other ARM vendors? I.e. move the selection of CMSIS in the modules' Kconfig?

Reply:
Technically, `select HAS_CMSIS_CORE` is not really needed anywhere other than in arch/arm/core/Kconfig because `CPU_CORTEX_M` already selects it.

I think these redundant selections could use a clean-up in the future (after all, since arch/arm/core relies on CMSIS-Core, so this makes sense).

Reply:
OH, I see, then could you clean them up in this PR, if possible?

Reply:
> OH, I see, then could you clean them up in this PR, if possible?

On second thought, one could probably argue that `HAS_CMSIS_CORE` should be selected by HAL modules because they actually use and rely on CMSIS-Core and, while it is true that such selections are redundant due to `CPU_CORTEX_M` selecting it already on Cortex-M HALs, purely from dependency perspective, `HAS_CMSIS_CORE` should be selected by them.

In response to that argument, I think, before doing this clean-up, we need to clarify that using `CPU_CORTEX_M` predicates the use of CMSIS-Core as this is an essential component which Cortex-M port cannot function without and therefore it is absolutely safe to assume that `CPU_CORTEX_M` will always select `HAS_CMSIS_CORE`.

If there's no counterargument to this, I add a commit for this clean-up tomorrow.

Reply:
All I want is consistency. Right now Nordic selects the CMSIS in the SOC definition, while other vendors select it in the module Kconfig. So, I think we can just make nordic look like the others.

I agree that HAS_CMSIS should still be part of the vendor module Kconfig; this actually signifies that the HAL headers _do_ use the CMSIS headers. In theory, this does not need to be the case.

Reply:
A major `HAS_CMSIS_CORE` usage clean-up involving redundant selection removal and/or other forms of refactoring can be done later in a separate PR (preferably after Cortex-R and Cortex-A port are added, as this would probably bring in quite a few big changes).

For now, I have added a `HAS_CMSIS_CORE` usage clean-up only for consistency.

Reply:
I like that the CMSIS_CORE is now selected in NRFX,instead of nRF SOC SERIES.